internal/pkg/pluginengine/delete.go

```diff
@@ -30,17 +30,18 @@ func Delete(fname string) {
 	p := planmanager.NewDeletePlan(smgr, cfg)
 	if len(p.Changes) == 0 {
 		log.Println("Nothing needs to be deleted. There is nothing to do.")
-		return
+		return nil
 	}
 
 	errsMap := execute(p)
-	if len(errsMap) == 0 {
-		log.Println("All plugins deleted successfully.")
-		return
+	if len(errsMap) != 0 {
```

Comment:
if len(errsMap) == 0 {
 	log.Println("All plugins deleted successfully.")
 	return nil
}

for k, e := range errsMap {
	log.Printf("%s -> %s", k, e)
}
	return errors.New("some error(s) occurred during plugins delete process")
 }

Reply:
This function is a bit long, so putting the "normal" return result at the end makes it easier to read. Otherwise, the reader might need to dig in the 30-line of code to see where the normal return is.